src/commands/run.rs

```diff
@@ -376,13 +391,54 @@ impl RunCommand {
         // Invoke the function and then afterwards print all the results that came
         // out, if there are any.
         let mut results = vec![Val::null(); ty.results().len()];
-        func.call(store, &values, &mut results).with_context(|| {
+        let invoke_res = func.call(store, &values, &mut results).with_context(|| {
             if let Some(name) = name {
                 format!("failed to invoke `{}`", name)
             } else {
                 format!("failed to invoke command default")
             }
-        })?;
+        });
+        match invoke_res {
+            Ok(_) => {}
+            Err(err) => {
```

Comment:
I think this logic below may want to be conditional on a `err.is::<wasmtime::Trap>()` condition since there's a number of other unrelated things that could go wrong during execution.